src/composables/transactionList.ts

```diff
@@ -0,0 +1,338 @@
+
+    if (transactionState.nextPageUrl === null && !recent) {
+      return [];
+    }
+    const { getMiddleware } = useMiddleware({ store });
+
+    const [
+      normal,
+      pending,
+      tip,
+    ] = await Promise.all([
+      fetchTransactionsFromMiddleware(transactionState, address, recent, limit),
+      fetchPendingTransactions(address),
+      fetchTipWithdrawnTransactions(address, recent),
+    ]);
+
+    const lastPendingTransaction = pending?.[pending.length - 1];
+    if (lastPendingTransaction?.type === AEX9_TRANSFER_EVENT) {
+      const middleware = await getMiddleware();
+      pending[pending.length - 1] = await middleware.getTx(lastPendingTransaction.payload.txHash);
+    }
+
+    let preparedTransactions = [
+      ...pending,
+      ...normal,
+    ]
+      .filter(({ type }) => !type?.startsWith('Internal'))
+      .map((transaction) => ({
+        ...(transaction.payload || transaction),
+        transactionOwner: address,
+        ...(transaction.type === AEX9_TRANSFER_EVENT
+          ? {
+            tx: {
+              ...transaction.payload,
+              callerId: transaction.payload.senderId,
+              type: 'ContractCallTx',
+            },
+            hash: transaction.payload.txHash,
+            incomplete: true,
+          } as ITransaction
+          : {}),
+      }));
+
+    preparedTransactions = uniqBy(preparedTransactions.reverse(), 'hash').reverse();
+    const minMicroTime = Math.min.apply(null, preparedTransactions.map((tx) => tx.microTime));
+    tip.forEach((f) => {
+      if (f.microTime
+        && (
+          minMicroTime < f.microTime
+          || (preparedTransactions.length === 0 && minMicroTime > f.microTime))
+      ) {
+        preparedTransactions.push({ ...f, transactionOwner: address });
+      }
+    });
+    preparedTransactions = orderBy(preparedTransactions, ['microTime'], ['desc']);
+
+    const oldPendingTransactionForAccount: ITransaction[] = (
+      transactions.value[address]?.pending[nodeNetworkId.value!] || []
+    );
+
+    oldPendingTransactionForAccount.forEach(({ hash }) => {
+      if (preparedTransactions.some((tx) => tx.hash === hash && !tx.pending)) {
+        removePendingTransactionByAccount(address, hash);
+      }
+    });
+
+    preparedTransactions = recent
+      ? preparedTransactions.slice(0, limit)
+      : preparedTransactions;
+
+    transactions.value[address].loaded = uniqBy(
+      [...transactions.value[address].loaded, ...preparedTransactions],
+      'hash',
+    );
+
+    return preparedTransactions;
+  }
+
+  watch(nodeNetworkId, (value, oldValue) => {
+    if (value) {
+      setLocalStorageItem([LOCAL_STORAGE_KEY, oldValue!], transactions.value);
+      transactions.value = getLocalStorageItem([LOCAL_STORAGE_KEY, value!]) || {};
+    }
+  });
+
+  watch(transactions, (value) => {
+    setLocalStorageItem([LOCAL_STORAGE_KEY, nodeNetworkId.value!], value);
+  }, { deep: true });
+
+  onMounted(async () => {
```

Comment:
When i used IIFE then getAeSdk not worked correctly, and application crashed then.

Reply:
What you mean by "not worked correctly"?

Reply:
await getAeSdk(); is never resolve.

Reply:
Probably because you use `useStore` from the `vuex` here: https://github.com/aeternity/superhero-wallet/pull/2171/files#diff-dbd73d02753926bab2746dfd799dec8c1b05c3920dfb52003d5f02bb3539521dR32.

Here's a topic on this: https://www.reddit.com/r/vuejs/comments/mki721/is_it_possible_to_access_usestore_from_another/